worlds/oot_soh/location_access/dungeons/ganons_castle.py

```diff
@@ -5,22 +5,396 @@
 
 
 class EventLocations(str, Enum):
-    GANON_DEFEATED = "Ganon Defeated",
+    GANONS_CASTLE_FREE_FAIRIES = "Ganon's Castle Free Fairies"
+    GANONS_CASTLE_FOREST_TRIAL_AREA = "Ganon's Castle Forest Trial Area"
+    GANONS_CASTLE_FIRE_TRIAL_AREA = "Ganon's Castle Fire Trial Area"
+    GANONS_CASTLE_WATER_TRIAL_AREA = "Ganon's Castle Water Trial Area"
+    GANONS_CASTLE_SHADOW_TRIAL_AREA = "Ganon's Castle Shadow Trial Area"
+    GANONS_CASTLE_SPIRIT_TRIAL_AREA = "Ganon's Castle Spirit Trial Area"
+    GANONS_CASTLE_LIGHT_TRIAL_AREA = "Ganon's Castle Light Trial Area"
+    GANONS_CASTLE_BLUE_FIRE_ACCESS = "Ganon's Castle Blue Fire Access"
+    GANONS_CASTLE_WATER_TRIAL_FAIRY_POT = "Ganon's Castle Water Trial Fairy Pot"
+    GANONS_CASTLE_SPIRIT_TRIAL_NUT_POT = "Ganon's Castle Spirit Trial Nut Pot"
+    GANON_DEFEATED = "Ganon Defeated"
+    
+class LocalEvents(str, Enum):
+    GANONS_CASTLE_FOREST_TRIAL_CLEARED = "Ganon's Castle Forest Trial Cleared"
+    GANONS_CASTLE_FIRE_TRIAL_CLEARED = "Ganon's Castle Fire Trial Cleared"
+    GANONS_CASTLE_WATER_TRIAL_CLEARED = "Ganon's Castle Water Trial Cleared"
+    GANONS_CASTLE_SHADOW_TRIAL_CLEARED = "Ganon's Castle Shadow Trial Cleared"
+    GANONS_CASTLE_SPIRIT_TRIAL_CLEARED = "Ganon's Castle Spirit Trial Cleared"
+    GANONS_CASTLE_LIGHT_TRIAL_CLEARED = "Ganon's Castle Light Trial Cleared"
+    
 
 
 def set_region_rules(world: "SohWorld") -> None:
-    player = world.player
+    ## Ganon's Castle Entryway
+    # Connections
+    connect_regions(Regions.GANONS_CASTLE_ENTRYWAY, world, [
+        (Regions.GANONS_CASTLE_LOBBY, lambda bundle: True),
+        (Regions.CASTLE_GROUNDS_FROM_GANONS_CASTLE, lambda bundle: True)
+    ])
     
-    # TODO: Temporary to test generation
-    connect_regions(Regions.KOKIRI_FOREST, world, [
-        (Regions.GANONS_ARENA, lambda bundle: True)
+    ## Ganon's Castle Lobby
+    # Connections
+    connect_regions(Regions.GANONS_CASTLE_LOBBY, world, [
+        (Regions.GANONS_CASTLE_ENTRYWAY, lambda bundle: True),
+        (Regions.GANONS_CASTLE_FOREST_TRIAL, lambda bundle: True),
+        (Regions.GANONS_CASTLE_FIRE_TRIAL, lambda bundle: True),
+        (Regions.GANONS_CASTLE_WATER_TRIAL, lambda bundle: True),
+        (Regions.GANONS_CASTLE_SHADOW_TRIAL, lambda bundle: True),
+        (Regions.GANONS_CASTLE_SPIRIT_TRIAL, lambda bundle: True),
+        (Regions.GANONS_CASTLE_LIGHT_TRIAL, lambda bundle: can_use(Items.GOLDEN_GAUNTLETS, bundle)),
+        (Regions.GANONS_TOWER_ENTRYWAY, lambda bundle: True),
+        (Regions.GANONS_CASTLE_DEKU_SCRUBS, lambda bundle: (can_do_trick(Tricks.LENS_GANON) or
+                                                            can_use(Items.LENS_OF_TRUTH, bundle))),
```

Comment:
```suggestion
        (Regions.GANONS_CASTLE_DEKU_SCRUBS, lambda bundle: can_do_trick(Tricks.LENS_GANON) or
                                                            can_use(Items.LENS_OF_TRUTH, bundle)),
```
Nitpick but these parenthesis are unnecessary.